### PR TITLE
Moves camera to pointer when ctrl space is stroke

### PR DESCRIPTION
**Pull request by thomas11111111**
"

### CODE REVIEW

**Comment by GoogleFrog** [comment]
I don't understand what this is for. It should at least be bindable in hotkeys.

**Comment by Licho1** [comment]
space is for queue insert and can be used with alt

**Comment by thomas11111111** [comment]
it can be used to move the camera around

**Comment by GoogleFrog** [comment]
The UI for this looks very idiosyncratic. Why does the mouse warp back to the screen centre? How accessible is Ctrl+Space?

File: Camera_to_Mouse.lua [hunk 1]
```diff
@@ -0,0 +1,51 @@
+-- Default widget stuff
+function widget:GetInfo()
+  return {
+    name      = "Camera to mouse-pointer",
+    desc      = "Camera moves to your mouse-pointer when alt+space is clicked",
+    author    = "Tumulten",
+    date      = "Mar 4, 2014",
+    license   = "GNU GPL, v2 or later",
+    layer     = 0,
+    enabled   = true
```
Comment (GoogleFrog) [inline Camera_to_Mouse.lua:10]: Should be disabled by default.

File: Camera_to_Mouse.lua [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	local altDown,ctrlDown,metaDown,shiftDown = Spring.GetModKeyState()
+	-- If ctrl and meta buttons are pressed down
+	if ctrlDown and metaDown then
+			-- We call function MoveCamera
+			MoveCamera()
+	end
+	-- Script ends
+end
+
+function MoveCamera()
```
Comment (GoogleFrog) [inline Camera_to_Mouse.lua:25]: Should be local

File: Camera_to_Mouse.lua [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		
+		if Spring.GetModUICtrl then
+			-- Get the screen resolution
+			local screenSizeX, screenSizeY = Spring.GetScreenGeometry() 
+			-- This will centre the mouse pointer back to the middle of the screen
+			Spring.WarpMouse(screenSizeX/2,screenSizeY/2)
+		end
+	end
+	-- Script goes back up to function KeyPress(the function that called this one)
+end
```
Comment (GoogleFrog) [inline Camera_to_Mouse.lua:51]: Missing newline.

File: Camera_to_Mouse.lua [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	local mouseX,mouseZ = Spring.GetMouseState()
+	-- Convert the mouse screen-coordinates into world-coordinates
+	local ignore,mouseWorldCord = Spring.TraceScreenRay(mouseX,mouseZ,true)
+	-- If the mouse is outside of the map Spring.TraceScreenRay will return nil so we have to check that it is not nil
+	if mouseWorldCord ~= nil then
+	-- Now the camera can be moved to the new location which is mouseWorldCord X, the cameras height from before (Y) and mouseWorldCord Z
+		Spring.SetCameraTarget(mouseWorldCord[1],camY,mouseWorldCord[3], 0)
+		
+		
+		if Spring.GetModUICtrl then
```
Comment (GoogleFrog) [inline Camera_to_Mouse.lua:43]: Redundant check

File: Camera_to_Mouse.lua [hunk 5]
```diff
@@ -0,0 +1,51 @@
+-- Default widget stuff
+function widget:GetInfo()
+  return {
+    name      = "Camera to mouse-pointer",
+    desc      = "Camera moves to your mouse-pointer when alt+space is clicked",
```
Comment (GoogleFrog) [inline Camera_to_Mouse.lua:5]: Description is wrong, and this is the only way people will learn what the widget does.

**Comment by Licho1** [comment]
other cameras use middle mouse  button

**Comment by thomas11111111** [comment]
redundancy -> closing

**Comment by GoogleFrog** [comment]
> redundancy -> closing

What? Are you always just going to close PRs if there is any stylistic feedback made on them? What is the purpose of this PR and what is your goal more generally?

**Comment by Licho1** [comment]
You can modify your commits in your branch and it will also modify PR. PR is not static so you can keep just single PR and work on it.

**Pull request closed without merging**